k4MarlinWrapper/src/components/EDM4hep2Lcio.cpp

```diff
@@ -0,0 +1,241 @@
+      lcio_vertex->setProbability( vertex.getProbability() );
+      lcio_vertex->setPosition( vertex.getPosition()[0], vertex.getPosition()[1], vertex.getPosition()[2]  );
+      lcio_vertex->setCovMatrix( vertex.getCovMatrix().data() );
+      lcio_recp->setStartVertex(lcio_vertex);
+
+      // Associated particle to the vertex
+      edm4hep::ConstReconstructedParticle vertex_rp = vertex.getAssociatedParticle();
+      if (vertex_rp.isAvailable()) {
+        // TODO add rp
+      }
+    }
+
+    // Tracks
+    for (int j=0; j < edm_rp.tracks_size(); ++j) {
+      edm4hep::ConstTrack edm_rp_tr = edm_rp.getTracks(j);
+      for (auto& track : lcio_tracks_vec) {
+        if (track.second == edm_rp_tr) {
+          lcio_recp->addTrack(track.first);
+        }
+      }
+    }
+
+    // Add converted LCIO RecoParticle, and associated EDM4hep RecoParticle
+    lcio_rec_particles_vec.emplace_back(
+      std::make_pair(lcio_recp, edm_rp)
+    );
+  }
+}
+
+
+void EDM4hep2LcioTool::convert(
+  const std::string& type,
+  const std::string& name)
+{
+  if (type == "edm4hep::TrackCollection") {
+    addLCIOConvertedTracks(lcio_tracks_vec, name);
+  } else
+  if (type == "edm4hep::ParticleIDCollection") {
+    addLCIOParticleIDs(lcio_particleIDs_vec, name);
+  } else
+  if (type == "edm4hep::ReconstructedParticleCollection") {
+    addLCIOReconstructedParticles(
+      lcio_rec_particles_vec,
+      lcio_particleIDs_vec,
+      lcio_tracks_vec,
+      name);
+  } else {
+    error() << "Error trying to convert requested " << type << " with name " << name << endmsg;
+  }
+}
+
+
+StatusCode EDM4hep2LcioTool::convertCollections(
+  const Gaudi::Property<std::vector<std::string>>& parameters,
+  lcio::LCEventImpl* lcio_event)
+{
+  info() << "Converting EDM4hep to LCIO requested collections. " << endmsg;
+
+  if (parameters.size() % 3 != 0) {
+    error() << " Error processing conversion parameters. 3 arguments per collection expected. " << endmsg;
+    return StatusCode::FAILURE;
+  }
+
+  for (int i = 0; i < parameters.size(); i=i+3) {
+    convert(parameters[i], parameters[i+1]);
+  }
+
+  // Populate LCIO event with EDM4hep info
+  auto* tracks = new lcio::LCCollectionVec(lcio::LCIO::TRACK);
+  auto* recops = new lcio::LCCollectionVec(lcio::LCIO::RECONSTRUCTEDPARTICLE);
+
+  // LCIO event to populate
+  lcio_event->setRunNumber(1);
+  const auto p1 = (std::chrono::system_clock::now().time_since_epoch()).count();
+  lcio_event->setTimeStamp(p1);
+
+  // Add converted tracks to LCIO vector
+  for (auto& tr : lcio_tracks_vec) {
+    tracks->addElement(tr.first);
+  }
+  lcio_event->addCollection(tracks, lcio::LCIO::TRACK);
+
+  // Add converted reconstructed particles to LCIO event
+  for (auto& rp : lcio_rec_particles_vec) {
+    recops->addElement(rp.first);
+  }
+  lcio_event->addCollection(recops, lcio::LCIO::RECONSTRUCTEDPARTICLE);
+
+  // Register LCIO event in TES
+  info() << "Registering converted EDM4hep to LCIO event in TES" << endmsg;
```

Comment:
Probably not neeed, can just assume that the LCIO event is in the data store (this must be handled somewhere else)